staging/src/k8s.io/code-generator/cmd/validation-gen/validation.go

```diff
@@ -854,6 +857,23 @@ func (g *genValidations) emitRegisterFunction(c *generator.Context, schemeRegist
 	sw.Do("}\n\n", nil)
 }
 
+// toResourceList returns a list of resources that are supported by a kind.
+func (g *genValidations) toResourceList(rootType *types.Type) []string {
+	supportedSubresources := supportedSubresourceTags(rootType)
+
+	if subresource, isSubresource := isSubresourceTag(rootType); isSubresource {
+		supportedSubresources.Insert(subresource)
+	} else {
+		supportedSubresources.Insert("/")
+	}
```

Comment:
For resources with `/status` we're expecting them to be both a subresource and spec, right?

Reply:
Correct.

The benefit having `+k8s:isSubresource` is that it makes it possible to mark Scale as being a dedicated Kind that is only for "/scale" subresources and not for any root "/" resources.

For "/status", the root resource should be tagged with "+k8s:supportsSubresource=/status", but `+k8s:isSubresource` should not be used, because there is no Go type that _is_ the scale subresource.